WordPress/wporg-repo-tools
# Add a basic block template for `mu-plugins` blocks

See #19.

This PR adds a template for generating a block to be used in [`wporg-mu-plugins`](https://github.com/WordPress/wporg-mu-plugins). The template can be used with `@wordpress/create-block`, and it will generate a block with the answers you give it.

The template is a forked version of the default `create-block` template — I've removed the overly-helpful comments, formatted the code with our code style, and removed the Sass from the `src` file. Now there's a separate `postcss` file which is where the CSS will live, as `*.pcss` to indicate that it's processed with PostCSS. Both `src` and `postcss` will be built down into a `build` folder (see https://github.com/WordPress/wporg-mu-plugins/pull/211 for the updated build process).

I kept this as basic as possible, so it really only provides a skeleton — if we find we need more structure after using it a bit, we can add it in to the template.

For an example, this will create a `new-block` folder with all the default values:

```bash
$ cd wporg-mu-plugins/mu-plugins/blocks
$ npx @wordpress/create-block -t "./vendor/wporg/wporg-repo-tools/templates/mu-block" new-block

Creating a new WordPress plugin in the "new-block" directory.

Creating a "block.json" file.

Creating a "package.json" file.

Done: WordPress plugin "New WordPress Block" bootstrapped in the "new-block" directory.

To enter the directory type:

  $ cd new-block

Code is Poetry
```

Now you have a new block, and you can open up the folder and see the generated code. The generated `README.md` has some instructions on how to finish setting this up in wporg-mu-plugins 

**To Test**

1. Check out this PR
2. Move to a directory where you can create a new folder
3. Run `npx @wordpress/create-block -t "/[SOME_PATH]/wporg-repo-tools/templates/mu-block"`
4. Answer the questions or leave them default
5. It should generate a folder
6. Open the new folder and read through the generated code
